plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/polaris/PolarisHiveMetastore.java

```diff
@@ -0,0 +1,575 @@
+
+    @Inject
+    public PolarisHiveMetastore(PolarisRestClient polarisClient)
+    {
+        this.polarisClient = requireNonNull(polarisClient, "polarisClient is null");
+    }
+
+    @Override
+    public Optional<Database> getDatabase(String databaseName)
+    {
+        try {
+            PolarisNamespace namespace = polarisClient.getNamespace(databaseName);
+            return Optional.of(Database.builder()
+                    .setDatabaseName(namespace.getName())
+                    .setParameters(namespace.getProperties())
+                    .build());
+        }
+        catch (PolarisNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to get database: " + databaseName, e);
+        }
+    }
+
+    @Override
+    public List<String> getAllDatabases()
+    {
+        try {
+            return polarisClient.listNamespaces(Optional.empty())
+                    .stream()
+                    .map(PolarisNamespace::getName)
+                    .collect(toImmutableList());
+        }
+        catch (Exception e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, "Failed to list databases", e);
+        }
+    }
+
+    @Override
+    public Optional<Table> getTable(String databaseName, String tableName)
+    {
+        // Try Iceberg tables first
+        try {
+            if (polarisClient.tableExists(databaseName, tableName)) {
+                PolarisTableMetadata metadata = polarisClient.loadTable(databaseName, tableName);
+                return Optional.of(convertIcebergToHiveTable(databaseName, tableName, metadata));
+            }
+        }
+        catch (Exception e) {
+            // Fall through to try generic tables
+        }
+
+        // Try generic tables
+        try {
+            if (polarisClient.genericTableExists(databaseName, tableName)) {
+                PolarisGenericTable genericTable = polarisClient.loadGenericTable(databaseName, tableName);
+                return Optional.of(convertGenericToHiveTable(databaseName, tableName, genericTable));
+            }
+        }
+        catch (Exception e) {
+            // Table not found
+        }
+
+        return Optional.empty();
+    }
+
+    public Set<ColumnStatisticType> getSupportedColumnStatistics(Type type)
+    {
+        return ImmutableSet.of();
+    }
+
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName)
+    {
+        return PartitionStatistics.empty();
+    }
+
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
+    {
+        return ImmutableMap.of();
+    }
+
+    @Override
+    public void updateTableStatistics(String databaseName, String tableName, OptionalLong acidWriteId, StatisticsUpdateMode mode, PartitionStatistics statisticsUpdate)
+    {
+        // Statistics updates not supported yet
```

Comment:
Throw an UnsupportedOperationException here.  You should nto get called unless you declare supported statistics.